tests/Models/Multiplayer/ScoreLinkTest.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace Tests\Models\Multiplayer;
+
+use App\Exceptions\InvariantException;
+use App\Models\Beatmap;
+use App\Models\Multiplayer\PlaylistItem;
+use App\Models\Multiplayer\ScoreLink;
+use App\Models\User;
+use Carbon\Carbon;
+use Tests\TestCase;
+
+class ScoreLinkTest extends TestCase
+{
+    public function testRequiredModsMissing()
+    {
+        $user = User::factory()->create();
+        $beatmap = Beatmap::factory()->create();
+        $playlistItem = PlaylistItem::factory()->create([
+            'required_mods' => [[
+                'acronym' => 'HD',
+            ]],
+        ]);
+        $scoreLink = ScoreLink::factory()->create([
```

Comment:
this should either be `->make` or `new ScoreLink([...])` because otherwise it'll create an already completed score which doesn't make sense in this context.

(looking again I should've changed the `complete` function as static method to prevent this from being allowed at all)